File: app/src/main/java/com/aivoice/translate/Activity/SplashActivity.java
package com.aivoice.translate.Activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;


import com.aivoice.translate.R;
import com.aivoice.translate.base.BaseActivity;
import com.aivoice.translate.utils.Utils;

import butterknife.BindView;

public class SplashActivity extends BaseActivity {
    @BindView(R.id.mIvLogo)
    ImageView mIvLogo;
    @BindView(R.id.mIvTitle)
    TextView mIvTitle;
    public static SplashActivity splashActivity;

    public static SplashActivity getInstance() {
        return splashActivity;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_splash;
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        splashActivity = this;


        mStartAct();
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    private void mStartAct() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (Utils.isNetworkAvailable(SplashActivity.this)) {
                    startActivity(new Intent(SplashActivity.this, IntroActivity.class));
                    finish();
                } else {
                    Toast.makeText(SplashActivity.this, "Turn On Internet..!", Toast.LENGTH_SHORT).show();
                }
            }
        }, 2000);

    }
}

File: app/src/main/java/com/texttranslate/voiceimage/SplashActivity.kt
package com.texttranslate.voiceimage

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.texttranslate.voiceimage.databinding.ActivitySplashBinding
import com.texttranslate.voiceimage.utils.Utils

@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {
    var splashActivity: SplashActivity? = null
    var binding: ActivitySplashBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashBinding.inflate(
            layoutInflater
        )
        setContentView(binding!!.getRoot())
        splashActivity = this
        Handler().postDelayed({
            if (Utils.isNetworkAvailable(this@SplashActivity)) {
                startActivity(Intent(this@SplashActivity, IntroActivity::class.java))
                finish()
            } else {
                Toast.makeText(this@SplashActivity, "Turn On Internet..!", Toast.LENGTH_SHORT)
                    .show()
            }
        }, 2000)
    }
}